### DISCUSSION TITLE
how to encode mesh in c++

### DISCUSSION THREAD

**Discussion by lihongyi1234** [zeux/meshoptimizer#308] [Q&A] [answered]
I'm confused wirh encode .obj file when I look at demo code,  where i can set compress params
![微信图片_20210701193556](https://user-images.githubusercontent.com/36233282/124118777-4a3dfc00-daa4-11eb-8701-899f541fc8b4.png)

**Comment by zeux** [comment]
What parameters are you asking about?

**Comment by lihongyi1234** [comment]
> What parameters are you asking about?

use N-bit uqntization for position/texture/normal parameters that encode in draco

**Comment by zeux** [comment, marked as answer]
Ah - that's a quantization step that is really up to the application to design/implement. The demo program implements two simple variants, see two `packMesh` functions in the source, but really it somewhat depends on the requirements - and some implementations may require additional shader decoding (e.g. the second packMesh variant uses an octahedral normal encoding that's optimal from the compression & quality perspective but requires a couple shader instructions to undo).
